airflow/models/taskinstance.py

```diff
@@ -70,6 +70,15 @@
 from airflow.utils.state import State
 from airflow.utils.timeout import timeout
 
+try:
+    from kubernetes.client.api_client import ApiClient
+
+    from airflow.executors.kubernetes_executor import KubeConfig, create_pod_id
+    from airflow.kubernetes.pod_generator import PodGenerator
+    from airflow.settings import pod_mutation_hook
```

Comment:
Do those imports have to be in try / except clause?

Reply:
No. It alway, bu we should use`from airflow import settings` to avoid cache.

Reply:
These ones do because they need the kubernetes library, which is not a requirement for airflow.